copycat-api/src/main/java/org/apache/kafka/copycat/storage/Converter.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package org.apache.kafka.copycat.storage;
+
+/**
+ * The Converter interface provides support for translating between Copycat's runtime data format
+ * and the "native" runtime format used by the serialization layer. This is used to translate
+ * two types of data: records and offsets. The (de)serialization is performed by a separate
```

Comment:
Yes, this really only applies to source offsets currently.

One thing that has bothered me is the inconsistency between how offsets are handled for source and sink tasks. I'm also wondering if there are any cases where storing additional data in a sink task offset commit would be useful. We could use the metadata field once that's exposed via the API for that, but we'd also have to make it available to Copycat somehow. In particular, I'm wondering if there is ever a case where you can get better delivery semantics to the destination system if you, e.g., get some ID back each time you send a message to it, and if you record that ID you might be able to get the destination system to do dedup -- however, I haven't come up with a system where this actually works yet :)

The other drawback to having them managed in different ways is that tooling needs to be different for source tasks vs. sink tasks. e.g., anything like burrow that tries to process the offsets topics will need to have two different implementations.